x-pack/platform/plugins/shared/alerting/server/alerts_service/lib/create_concrete_write_index.ts

```diff
@@ -206,3 +236,54 @@ export async function setConcreteWriteIndex(opts: SetConcreteWriteIndexOpts) {
     );
   }
 }
+
+const increaseFiledsLimit = async ({
+  err,
+  esClient,
+  concreteIndexInfo,
+  logger,
+  increment,
+}: {
+  err: Error;
+  esClient: ElasticsearchClient;
+  concreteIndexInfo: ConcreteIndexInfo;
+  logger: Logger;
+  increment: number;
+}): Promise<number | undefined> => {
+  const { alias } = concreteIndexInfo;
+  const match = err.message
+    ? err.message.match(/Limit of total fields \[(\d+)\] has been exceeded/)
+    : null;
+
+  if (match !== null) {
+    const exceededLimit = parseInt(match[1], 10);
+    const newLimit = exceededLimit + increment;
+
+    const { index_templates: indexTemplates } = await retryTransientEsErrors(
+      () =>
+        esClient.indices.getIndexTemplate({
+          name: `${alias}-index-template`,
+        }),
+      { logger }
+    );
+    const template = indexTemplates[0];
```

Comment:
is it possible this returns an empty array? maybe add a length check?

Reply:
added a length check